tests/Integration/Behaviour/Features/Scenario/Product/update_seo.feature

```diff
@@ -0,0 +1,168 @@
+  As an employee
+  I need to be able to update product SEO options from Back Office
+
+  Scenario: I update product SEO
+    Given I add product "product1" with following information:
+      | name       | en-US: just boots         |
+      | is_virtual | false                     |
+    Given I add product "product2" with following information:
+      | name       | en-US: just boots         |
+      | is_virtual | false                     |
+    And product product2 should have following values:
+      | redirect_type    | 404                 |
+    And product product2 should not have a redirect target
+    And product "product2" localized "meta_title" is "en-US:"
+    And product "product2" localized "meta_description" is "en-US:"
+    And product "product2" localized "link_rewrite" is "en-US:"
+    When I update product product2 SEO information with following values:
+      | meta_title       | en-US:product2 meta title       |
+      | meta_description | en-US:product2 meta description |
+      | link_rewrite     | en-US:waterproof-boots          |
+      | redirect_type    | 301-product                     |
+      | redirect_target  | product1                        |
+    Then product "product2" localized "meta_title" should be "en-US:product2 meta title"
+    And product "product2" localized "meta_description" should be "en-US:product2 meta description"
+    And product "product2" localized "link_rewrite" should be "en-US:waterproof-boots"
+    And product product2 should have following values:
+      | redirect_type    | 301-product               |
+    And product product2 redirect target should be product1
+
+  Scenario: Update product redirect type without providing redirect target
+    Given product "product2" localized "meta_title" should be "en-US:product2 meta title"
+    And product "product2" localized "meta_description" should be "en-US:product2 meta description"
+    And product "product2" localized "link_rewrite" should be "en-US:waterproof-boots"
+    And product product2 should have following values:
+      | redirect_type    | 301-product               |
+    And product product2 redirect target should be product1
+    When I update product product2 SEO information with following values:
+      | redirect_type    | 302-product               |
+      | redirect_target  |                           |
+    Then I should get error that product redirect_target is invalid
+    And product "product2" localized "meta_title" should be "en-US:product2 meta title"
+    And product "product2" localized "meta_description" should be "en-US:product2 meta description"
+    And product "product2" localized "link_rewrite" should be "en-US:waterproof-boots"
+    And product product2 should have following values:
+      | redirect_type    | 301-product                |
+    And product product2 redirect target should be product1
+    When I update product product2 SEO information with following values:
```

Comment:
I believe you didn't test any scenario where you set a category ID

This makes me think about other more complex/advanced scenario and use cases:

I have a product redirected to a specified product, I change it to redirect category without specifying a category ID:
- should we reset the redirect target as it was targeting a product?
- should we keep the initial ID even if it was previously a product ID and now points to a category ID?

More generically, should the handler check that the provided IDs (product or category), are valid ones (existing and active)?

Reply:
Mmm, actually I believe the fact that we have a "double setter" manages the first question: `setRedirectOption` forces you to specify a target ID, so it will necessarily be overridden

The question about checking if IDs are valid in the handler remains though

Reply:
To update the option it is required to provide both values, which was the purpose of this VO (ensure that it is the expected intention to change both values because they are related).
I've added asserts to check if the entity exists depending on the redirect type :+1: